### PR TITLE
test(metrics/prometheus): refactor scrape in prometheus test

### PR DESCRIPTION
**Pull request by andynog**
close: #4133

### CODE REVIEW

File: libs/metrics/prometheus/prometheus_test.go [hunk 1]
```diff
@@ -248,3 +226,22 @@ func TestInconsistentLabelCardinality(t *testing.T) {
 		"a", "1", "b", "2", "c", "KABOOM!",
 	).Add(123)
 }
+
+func spinUpPrometheusServer() *httptest.Server {
+	s := httptest.NewServer(promhttp.HandlerFor(stdprometheus.DefaultGatherer, promhttp.HandlerOpts{}))
+
+	// minimal delay to allow the prometheus server come up with results and avoid errors during test
```
Comment (melekes) [inline libs/metrics/prometheus/prometheus_test.go:233]: What results? At this point, we have not registered any metrics.

Comment (andynog) [reply libs/metrics/prometheus/prometheus_test.go:233]: I think you are asking only about the comment right ? I meant just for the server to be up, not results, I can remove that from the comment

File: libs/metrics/prometheus/prometheus_test.go [hunk 2]
```diff
@@ -37,21 +37,16 @@ import (
...
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/stretchr/testify/require"
 
 	"github.com/cometbft/cometbft/libs/metrics/teststat"
 )
 
 func TestCounter(t *testing.T) {
-	s := httptest.NewServer(promhttp.HandlerFor(stdprometheus.DefaultGatherer, promhttp.HandlerOpts{}))
+	s := spinUpPrometheusServer()
+	require.NotNil(t, s)
```
Comment (melekes) [inline libs/metrics/prometheus/prometheus_test.go:47]: I don't think this is necessary.

File: libs/metrics/prometheus/prometheus_test.go [hunk 3]
```diff
@@ -82,16 +77,10 @@ func TestCounter(t *testing.T) {
 }
 
 func TestGauge(t *testing.T) {
-	s := httptest.NewServer(promhttp.HandlerFor(stdprometheus.DefaultGatherer, promhttp.HandlerOpts{}))
+	s := spinUpPrometheusServer()
+	require.NotNil(t, s)
```
Comment (melekes) [inline libs/metrics/prometheus/prometheus_test.go:81]: same

File: libs/metrics/prometheus/prometheus_test.go [hunk 4]
```diff
@@ -159,16 +143,10 @@ func TestHistogram(t *testing.T) {
 	// limit. That is, the count monotonically increases over the buckets. This
 	// requires a different strategy to test.
 
-	s := httptest.NewServer(promhttp.HandlerFor(stdprometheus.DefaultGatherer, promhttp.HandlerOpts{}))
+	s := spinUpPrometheusServer()
+	require.NotNil(t, s)
```
Comment (melekes) [inline libs/metrics/prometheus/prometheus_test.go:147]: same

**Review by melekes — APPROVED** [review]

**Pull request merged**